datalad/support/gitrepo.py

```diff
@@ -0,0 +1,63 @@
+"""
```

Comment:
All files should have the same header (yet to introduce throughout -- feel welcome to replace my custom headers everywhere in a separate PR):

```
# emacs: -*- mode: python-mode; py-indent-offset: 4; tab-width: 4; indent-tabs-mode: nil -*-
# ex: set sts=4 ts=4 sw=4 noet:
### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
#
#   See COPYING file distributed along with the datalad package for the
#   copyright and license terms.
#
### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
```